### PR TITLE
fix(search-engine): #MC-264 allow research in Calendar

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
- Match field change from verison 0.7.0 : field "calendar" in an event is no longer a String but an array
- Changes in i18n

## Checklist tests
try to research an event, you should get the info concerning it

## Issue ticket number and link
[MC-264](https://jira.support-ent.fr/browse/MC-264)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ private JsonArray formatSearchResult(final JsonArray results, final JsonArray co
 			final JsonObject j = results.getJsonObject(i);
 			final JsonObject jr = new JsonObject();
 			if (j != null) {
-				final String idCalendar = j.getString("calendar");
+				final String idCalendar = j.getJsonArray("calendar", new JsonArray()).getString(0);
```
Comment (sjollois) [inline src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java:130]: verif par rapport au size du JsonArray de "calendar" pour ne pas avoir d'exception

**Review by sjollois — CHANGES_REQUESTED** [review]

File: src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java (hunk 1 above)
Comment (BastienBenard) [reply src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java:130]: MagicString

File: src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java [hunk 2]
```diff
@@ -127,7 +127,7 @@ private JsonArray formatSearchResult(final JsonArray results, final JsonArray co
...
 				jr.put(aHeader.get(0), mapIdTitle.get(idCalendar));
 				jr.put(aHeader.get(1), formatDescription(locale, j.getString("title"), j.getString("description", ""),
 						j.getString("location", ""), j.getString("startMoment"), j.getString("endMoment")));
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java:133]: Magic String

File: src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java [hunk 3]
```diff
@@ -155,21 +155,21 @@ private String formatDescription(final String locale, final String title, final
...
-			descriptionRes = i18n.translate("calendar.search.description.location", locale, title, sDateRes,
+			descriptionRes = i18n.translate("calendar.search.description.location", "default-domain", locale, title, sDateRes,
 					eDateRes, location);
 		} else if (!description.isEmpty()){
-			descriptionRes = i18n.translate("calendar.search.description.desc", locale, title, sDateRes,
+			descriptionRes = i18n.translate("calendar.search.description.desc", "default-domain", locale, title, sDateRes,
 					eDateRes, description);
 		} else {
-			descriptionRes = i18n.translate("calendar.search.description.min", locale, title, sDateRes,
+			descriptionRes = i18n.translate("calendar.search.description.min", "default-domain", locale, title, sDateRes,
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/event/CalendarSearchingEvents.java:172]: "default-domain" MagicString

**Review by BastienBenard — COMMENTED** [review]
Est ce que tu peux faire un test sur la fonction formatSearchResult ?

**Review by aymericdevriesere — APPROVED** [review]

**Review by sjollois — APPROVED** [review]

File: src/test/java/net/atos/entng/calendar/event/CalendarSearchingEventsTest.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        JsonArray resultArray = Whitebox.invokeMethod(new CalendarSearchingEvents(), "formatSearchResult",
+                results, columnsHeader, mapIdTitle, locale);
+        ctx.assertEquals(expectedArray.toString(), resultArray.toString());
+
+//        Mockito.doAnswer(invocation -> {
+//            JsonArray resultMessage = invocation.getArgument(0);
+//            ctx.assertEquals(expectedArray, resultMessage);
+//            return null;
+//        }).when(i18n).translate(Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/event/CalendarSearchingEventsTest.java:90]: c'est utile ?

File: src/test/java/net/atos/entng/calendar/event/CalendarSearchingEventsTest.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Before
+    public void setUp(TestContext context) {
+        Whitebox.setInternalState(CalendarSearchingEvents.class, "i18n", i18n);
+    }
+
+    @Test
+    public void testFormatSearchResult_shouldReturnDefaultObjectArray (TestContext ctx) throws Exception {
+
+//"modified":{"$date":1669039699414}}]
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/event/CalendarSearchingEventsTest.java:45]: c'est utile ?

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**